---
title: "20210412-figures"
author: "Sujatha Jagannathan"
date: "4/12/2021"
output: html_document
---

### R set up
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library("readr")
library(tidyverse)
library("viridis")
library("RColorBrewer")
library("DT")
library("DataExplorer")
library("GGally")
library("ggthemes")
library("gridExtra")
library("ggfortify")
library("matrixStats")
library("pheatmap")
library("corrplot")
library("cowplot")
library(readxl)
library("ggpubr")

mytheme <- theme_few(base_size = 14) + 
                theme(
                  #aspect.ratio = 0.5, 
                  #legend.position = "none", 
                      panel.background = 
                          element_rect(
                              colour = "black", 
                              size=1))

# colors: https://bhaskarvk.github.io/colormap/ "cividis"

# colors
# dark_blue: "#042333ff"
# orange: "#f9a242ff"
# yellow: "#efe350ff"

# ff -> b2 for less opacity

```

### Data input
```{r}
deseq2_foldchange <- read_csv("data/deseq2-foldchange.csv")

yao_dux4_target_ddu251supp_table1 <- read_excel("data/yao-dux4-target-ddu251supp_table1.xls", skip = 1)

target_genes <- yao_dux4_target_ddu251supp_table1[, c(1:2)]

```


##### Figure 2
```{r}
data <- deseq2_foldchange[, c(2:54)]

data_rna_long <- data %>% 
  filter(gene_biotype == "protein_coding") %>% 
  dplyr::select(ensembl_gene_id, gene_name = external_gene_name, baseMean_RNA, log2FoldChange.04h_RNA, log2FoldChange.08h_RNA, log2FoldChange.14h_RNA, padj.04h_RNA, padj.08h_RNA, padj.14h_RNA)  %>% 
  unite(log2FoldChange.04h_RNA, padj.04h_RNA, col = "4h", sep = "_") %>% 
  unite(log2FoldChange.08h_RNA, padj.08h_RNA, col = "8h", sep = "_") %>% 
  unite(log2FoldChange.14h_RNA, padj.14h_RNA, col = "14h", sep = "_") %>% 
  pivot_longer(cols = `4h`:`14h`, names_to = "time", values_to = "log2FoldChange_padj") %>% 
  separate(log2FoldChange_padj, sep = "_", into = c("log2FoldChange", "padj")) %>%
  mutate(target_status = ifelse(ensembl_gene_id %in% target_genes$Ensembl, "true", "false")) %>%
  mutate_at(vars(starts_with("log2")),funs(as.numeric)) %>%
  mutate_at(vars(starts_with("padj")),funs(as.numeric)) %>% 
  dplyr::arrange(target_status) %>%
  mutate(significance = ifelse(padj < 0.01, "true", "false")) %>%
  mutate(up_down = ifelse(log2FoldChange > 0, "up", "down"))

# Fig 2C - MA plot

colsa<-alpha(c("dark grey", "#042333"),c(.2,.7))
#names(colsa)<-c("target","non-target")

ggplot(data_rna_long, 
            aes(
              x = log10(baseMean_RNA), 
              y = log2FoldChange)) + 
  geom_point(aes(fill = target_status, color = target_status, pch = significance), size = 2)+ 
  facet_wrap(~as_factor(time)) + 
  scale_fill_manual(values = colsa,"DUX4 target status") +
    scale_color_manual(values = colsa,"DUX4 target status") +
  scale_shape_manual(values = c(1, 21)) + 
  #theme(aspect.ratio = 1) +
  #ggtitle("DUX4") + 
  xlab(label = "log10 base expression") + 
  ylab(label = "log2 fold change DUX4/control")  + mytheme + geom_hline(yintercept=0, linetype="dashed", color = "black")

# number of significantly changed genes

temp <- data_rna_long %>% filter(significance == "true")
table(temp$time, temp$up_down)

  #     down   up
  # 14h 4080 4350
  # 4h   813  861
  # 8h  3077 3060
  
# colors
# dark_blue: "#042333ff"
# orange: "#f9a242ff"
# yellow: "#efe350ff"

# gene set analysis
# source of gene sets: https://www.gsea-msigdb.org/gsea/msigdb/search.jsp

data <- deseq2_foldchange[, c(2:54)]

# ubiquitin genes
ubiquitin_genes <- read_csv("data/GOBP_PROTEIN_POLYUBIQUITINATION.txt", col_names = FALSE)
names(ubiquitin_genes) <- c("gene_name")

# ubquitin_data <- data %>% filter( external_gene_name %in% ubiquitin_genes$gene_name) %>% select(ensembl_gene_id, external_gene_name, log2FoldChange.04h_RNA, log2FoldChange.08h_RNA, log2FoldChange.14h_RNA, padj.04h_RNA, padj.08h_RNA, padj.14h_RNA) %>% filter((padj.04h_RNA < 0.01 & abs(log2FoldChange.04h_RNA) > 1) | (padj.08h_RNA < 0.01 & abs(log2FoldChange.08h_RNA) > 1) | (padj.14h_RNA < 0.01 & abs(log2FoldChange.14h_RNA) > 1)) %>% unique()

ubiquitin_data <- data %>% filter( external_gene_name %in% ubiquitin_genes$gene_name) %>% dplyr::select(ensembl_gene_id, external_gene_name, log2FoldChange.04h_RNA, log2FoldChange.08h_RNA, log2FoldChange.14h_RNA, padj.04h_RNA, padj.08h_RNA, padj.14h_RNA) %>% filter((padj.04h_RNA < 0.01 & abs(log2FoldChange.04h_RNA) > 1)) %>% unique()

ubiquitin_data <- ubiquitin_data[complete.cases(ubiquitin_data), ]

# splicing genes
splicing_genes <- read_csv("data/GOBP_RNA_SPLICING.txt", col_names = FALSE)
names(splicing_genes) <- c("gene_name")

splicing_data <- data %>% filter( external_gene_name %in% splicing_genes$gene_name) %>% dplyr::select(ensembl_gene_id, external_gene_name, log2FoldChange.04h_RNA, log2FoldChange.08h_RNA, log2FoldChange.14h_RNA, padj.04h_RNA, padj.08h_RNA, padj.14h_RNA) %>% filter((padj.04h_RNA < 0.01 & abs(log2FoldChange.04h_RNA) > 1)) %>% unique()
splicing_data <- splicing_data[complete.cases(splicing_data), ]

# RNA processing genes
rna_processing_genes <- read_csv("data/GOBP_MRNA_PROCESSING.txt", col_names = FALSE)
names(rna_processing_genes) <- c("gene_name")

rna_processing_data <- data %>% filter( external_gene_name %in% rna_processing_genes$gene_name) %>% dplyr::select(ensembl_gene_id, external_gene_name, log2FoldChange.04h_RNA, log2FoldChange.08h_RNA, log2FoldChange.14h_RNA, padj.04h_RNA, padj.08h_RNA, padj.14h_RNA) %>% filter((padj.04h_RNA < 0.01 & abs(log2FoldChange.04h_RNA) > 1)) %>% unique()
rna_processing_data <- rna_processing_data[complete.cases(rna_processing_data), ]

# WNT signaling genes
wnt_signaling_genes <- read_csv("data/GOBP_CELL_CELL_SIGNALING_BY_WNT.txt", col_names = FALSE)
names(wnt_signaling_genes) <- c("gene_name")

wnt_signaling_data <- data %>% filter( external_gene_name %in% wnt_signaling_genes$gene_name) %>% dplyr::select(ensembl_gene_id, external_gene_name, log2FoldChange.04h_RNA, log2FoldChange.08h_RNA, log2FoldChange.14h_RNA, padj.04h_RNA, padj.08h_RNA, padj.14h_RNA) %>% filter((padj.04h_RNA < 0.01 & abs(log2FoldChange.04h_RNA) > 1)) %>% unique()
wnt_signaling_data <- wnt_signaling_data[complete.cases(wnt_signaling_data), ]

# ISR genes
isr_genes <- read_csv("data/GOBP_INTEGRATED_STRESS_RESPONSE_SIGNALING.txt", col_names = FALSE)
names(isr_genes) <- c("gene_name")

isr_data <- data %>% filter( external_gene_name %in% isr_genes$gene_name) %>% dplyr::select(ensembl_gene_id, external_gene_name, log2FoldChange.04h_RNA, log2FoldChange.08h_RNA, log2FoldChange.14h_RNA, padj.04h_RNA, padj.08h_RNA, padj.14h_RNA) %>% filter((padj.04h_RNA < 0.01 & abs(log2FoldChange.04h_RNA) > 1)) %>% unique()
isr_data <- isr_data[complete.cases(isr_data), ]

# anoikis genes
anoikis_genes <- read_csv("data/GOBP_ANOIKIS.txt", col_names = FALSE)
names(anoikis_genes) <- c("gene_name")

anoikis_data <- data %>% filter( external_gene_name %in% anoikis_genes$gene_name) %>% dplyr::select(ensembl_gene_id, external_gene_name, log2FoldChange.04h_RNA, log2FoldChange.08h_RNA, log2FoldChange.14h_RNA, padj.04h_RNA, padj.08h_RNA, padj.14h_RNA) %>% filter((padj.04h_RNA < 0.01 & abs(log2FoldChange.04h_RNA) > 1)) %>% unique()
anoikis_data <- anoikis_data[complete.cases(anoikis_data), ]

# Apoptosis genes
apoptosis_genes <- read_csv("data/GOBP_INTRINSIC_APOPTOTIC_SIGNALING_PATHWAY.txt", col_names = FALSE)
names(apoptosis_genes) <- c("gene_name")

apoptosis_data <- data %>% filter( external_gene_name %in% apoptosis_genes$gene_name) %>% dplyr::select(ensembl_gene_id, external_gene_name, log2FoldChange.04h_RNA, log2FoldChange.08h_RNA, log2FoldChange.14h_RNA, padj.04h_RNA, padj.08h_RNA, padj.14h_RNA) %>% filter((padj.04h_RNA < 0.01 & abs(log2FoldChange.04h_RNA) > 1)) %>% unique()
apoptosis_data <- apoptosis_data[complete.cases(apoptosis_data), ]

# muscle genes
muscle_genes <- read_csv("data/GOBP_MUSCLE_FIBER_DEVELOPMENT.txt", col_names = FALSE)
names(muscle_genes) <- c("gene_name")

muscle_data <- data %>% filter( external_gene_name %in% muscle_genes$gene_name) %>% dplyr::select(ensembl_gene_id, external_gene_name, log2FoldChange.04h_RNA, log2FoldChange.08h_RNA, log2FoldChange.14h_RNA, padj.04h_RNA, padj.08h_RNA, padj.14h_RNA) %>% filter((padj.04h_RNA < 0.01 & abs(log2FoldChange.04h_RNA) > 1)) %>% unique()
muscle_data <- muscle_data[complete.cases(muscle_data), ]

# oxidative stress genes
oxidative_stress_genes <- read_csv("data/GOBP_RESPONSE_TO_OXIDATIVE_STRESS.txt", col_names = FALSE)
names(oxidative_stress_genes) <- c("gene_name")

oxidative_stress_data <- data %>% filter( external_gene_name %in% oxidative_stress_genes$gene_name) %>% dplyr::select(ensembl_gene_id, external_gene_name, log2FoldChange.04h_RNA, log2FoldChange.08h_RNA, log2FoldChange.14h_RNA, padj.04h_RNA, padj.08h_RNA, padj.14h_RNA) %>% filter((padj.04h_RNA < 0.01 & abs(log2FoldChange.04h_RNA) > 1)) %>% unique()
oxidative_stress_data <- oxidative_stress_data[complete.cases(oxidative_stress_data), ]

# MAPK genes
mapk_genes <- read_csv("data/KEGG_MAPK_SIGNALING_PATHWAY.txt", col_names = FALSE)
names(mapk_genes) <- c("gene_name")

mapk_data <- data %>% filter( external_gene_name %in% mapk_genes$gene_name) %>% dplyr::select(ensembl_gene_id, external_gene_name, log2FoldChange.04h_RNA, log2FoldChange.08h_RNA, log2FoldChange.14h_RNA, padj.04h_RNA, padj.08h_RNA, padj.14h_RNA) %>% filter((padj.04h_RNA < 0.01 & abs(log2FoldChange.04h_RNA) > 1)) %>% unique()
mapk_data <- mapk_data[complete.cases(mapk_data), ]

# viral response genes
viral_response_genes <- read_csv("data/GOBP_DEFENSE_RESPONSE_TO_VIRUS.txt", col_names = FALSE)
names(viral_response_genes) <- c("gene_name")

viral_response_data <- data %>% filter( external_gene_name %in% viral_response_genes$gene_name) %>% dplyr::select(ensembl_gene_id, external_gene_name, log2FoldChange.04h_RNA, log2FoldChange.08h_RNA, log2FoldChange.14h_RNA, padj.04h_RNA, padj.08h_RNA, padj.14h_RNA) %>% filter((padj.04h_RNA < 0.01 & abs(log2FoldChange.04h_RNA) > 1)) %>% unique()
viral_response_data <- viral_response_data[complete.cases(viral_response_data), ]

# cell cycle genes
cell_cycle_genes <- read_csv("data/GOBP_CELL_CYCLE_PHASE_TRANSITION.txt", col_names = FALSE)
names(cell_cycle_genes) <- c("gene_name")

cell_cycle_data <- data %>% filter( external_gene_name %in% cell_cycle_genes$gene_name) %>% dplyr::select(ensembl_gene_id, external_gene_name, log2FoldChange.04h_RNA, log2FoldChange.08h_RNA, log2FoldChange.14h_RNA, padj.04h_RNA, padj.08h_RNA, padj.14h_RNA) %>% filter((padj.04h_RNA < 0.01 & abs(log2FoldChange.04h_RNA) > 1)) %>% unique()
cell_cycle_data <- cell_cycle_data[complete.cases(cell_cycle_data), ]

# G2-M genes
g2m_genes <- read_csv("data/GOBP_CELL_CYCLE_G2_M_PHASE_TRANSITION.txt", col_names = FALSE)
names(g2m_genes) <- c("gene_name")

g2m_data <- data %>% filter( external_gene_name %in% g2m_genes$gene_name) %>% dplyr::select(ensembl_gene_id, external_gene_name, log2FoldChange.04h_RNA, log2FoldChange.08h_RNA, log2FoldChange.14h_RNA, padj.04h_RNA, padj.08h_RNA, padj.14h_RNA) %>% filter((padj.04h_RNA < 0.01 & abs(log2FoldChange.04h_RNA) > 1)) %>% unique()
g2m_data <- g2m_data[complete.cases(g2m_data), ]

# G1-S genes
g1s_genes <- read_csv("data/GOBP_CELL_CYCLE_G1_S_PHASE_TRANSITION.txt", col_names = FALSE)
names(g1s_genes) <- c("gene_name")

g1s_data <- data %>% filter( external_gene_name %in% g1s_genes$gene_name) %>% dplyr::select(ensembl_gene_id, external_gene_name, log2FoldChange.04h_RNA, log2FoldChange.08h_RNA, log2FoldChange.14h_RNA, padj.04h_RNA, padj.08h_RNA, padj.14h_RNA) %>% filter((padj.04h_RNA < 0.01 & abs(log2FoldChange.04h_RNA) > 1)) %>% unique()
g1s_data <- g1s_data[complete.cases(g1s_data), ]

# actin genes
actin_genes <- read_csv("data/GOBP_ACTIN_FILAMENT_ORGANIZATION.txt", col_names = FALSE)
names(actin_genes) <- c("gene_name")

actin_data <- data %>% filter( external_gene_name %in% actin_genes$gene_name) %>% dplyr::select(ensembl_gene_id, external_gene_name, log2FoldChange.04h_RNA, log2FoldChange.08h_RNA, log2FoldChange.14h_RNA, padj.04h_RNA, padj.08h_RNA, padj.14h_RNA) %>% filter((padj.04h_RNA < 0.01 & abs(log2FoldChange.04h_RNA) > 1)) %>% unique()
actin_data <- actin_data[complete.cases(actin_data), ]

# ECM genes
ecm_genes <- read_csv("data/GOMF_EXTRACELLULAR_MATRIX_STRUCTURAL_CONSTITUENT.txt", col_names = FALSE)
names(ecm_genes) <- c("gene_name")

ecm_data <- data %>% filter( external_gene_name %in% ecm_genes$gene_name) %>% dplyr::select(ensembl_gene_id, external_gene_name, log2FoldChange.04h_RNA, log2FoldChange.08h_RNA, log2FoldChange.14h_RNA, padj.04h_RNA, padj.08h_RNA, padj.14h_RNA) %>% filter((padj.04h_RNA < 0.01 & abs(log2FoldChange.04h_RNA) > 1)) %>% unique()
ecm_data <- ecm_data[complete.cases(ecm_data), ]

# heatmaps
mat_breaks = seq(-5, 5, by = 0.5)

mat <- isr_data[3:5] #15
rownames(mat) <- isr_data$external_gene_name

mat <- anoikis_data[3:5] #19
rownames(mat) <- anoikis_data$external_gene_name

mat <- muscle_data[3:5] #24
rownames(mat) <- muscle_data$external_gene_name

mat <- viral_response_data[3:5] #100
rownames(mat) <- viral_response_data$external_gene_name

mat <- mapk_data[3:5] #110
rownames(mat) <- mapk_data$external_gene_name

mat <- apoptosis_data[3:5] #132
rownames(mat) <- apoptosis_data$external_gene_name

mat <- ubiquitin_data[3:5] #159
rownames(mat) <- ubiquitin_data$external_gene_name

mat <- oxidative_stress_data[3:5] #164
rownames(mat) <- oxidative_stress_data$external_gene_name

mat <- splicing_data[3:5] #191
rownames(mat) <- splicing_data$external_gene_name

mat <- rna_processing_data[3:5] #212
rownames(mat) <- rna_processing_data$external_gene_name

mat <- wnt_signaling_data[3:5] #223
rownames(mat) <- wnt_signaling_data$external_gene_name

mat <- cell_cycle_data[3:5] #223
rownames(mat) <- cell_cycle_data$external_gene_name

mat <- g1s_data[3:5] #223
rownames(mat) <- g1s_data$external_gene_name

mat <- g2m_data[3:5] #223
rownames(mat) <- g2m_data$external_gene_name

mat <- actin_data[3:5] #223
rownames(mat) <- actin_data$external_gene_name

mat <- ecm_data[3:5] #223
rownames(mat) <- ecm_data$external_gene_name

# plotting the heatmaps one gene set at a time
set.seed(123)
pheatmap(mat,
color = cividis(length(mat_breaks) - 1),
breaks = mat_breaks, 
clustering_method = "complete", #specify clustering method
border_color = "black",
na_col = "grey90", 
cluster_rows = T,
cluster_cols = F,
legend=F, 
treeheight_row = 0, 
labels_col = "", 
cellwidth = 12, cellheight = 12)

# Fig 2D - Cluster analysis
data <- deseq2_foldchange[, c(2:54)]

# significant at any time point
sig_any <- data %>% filter(gene_biotype == "protein_coding") %>% dplyr::select(ensembl_gene_id, external_gene_name, log2FoldChange.04h_RNA, log2FoldChange.08h_RNA, log2FoldChange.14h_RNA, padj.04h_RNA, padj.08h_RNA, padj.14h_RNA) %>% filter((padj.04h_RNA < 0.01 & abs(log2FoldChange.04h_RNA) > 1) | (padj.08h_RNA < 0.01 & abs(log2FoldChange.08h_RNA) > 1) | (padj.14h_RNA < 0.01 & abs(log2FoldChange.14h_RNA) > 1)) %>% unique()
sig_any <- sig_any[complete.cases(sig_any), ]
nrow(sig_any)
#6109

# calculating # of clusters by elbow method
# any
set.seed(20)
k.max <- 15
sig_any_cluster <- sig_any[,3:5]
wss <- sapply(1:k.max,
function(k){kmeans(sig_any_cluster, k, nstart=50,iter.max = 15 )$tot.withinss})

plot(1:k.max, wss,
type="b", pch = 19, frame = FALSE,
xlab="Number of clusters K",
ylab="Total within-clusters sum of squares - any")

# plotting k-means clusters

mat_breaks = seq(-5, 5, by = 0.5)

set.seed(123)
sig_any_cluster_kmeans <- pheatmap(
  mat = sig_any_cluster, 
  color = cividis(22), 
  breaks = mat_breaks, 
  border_color = "black",
  kmeans_k = 5, 
  cluster_cols = FALSE)

# trying to rename/reorder the clusters to swap the order of 3 and 2. 
# sig_any_cluster_kmeans$tree_row$order <- c(1, 3, 2, 4, 5)

# Note: the above did not work. I am manually changing cluster 2 to be cluster 3 and vice versa to keep the narrative less confusing. 

sig_any_cluster_kmeans$kmeans$centers

# 5 clusters
#   log2FoldChange.04h_RNA log2FoldChange.08h_RNA log2FoldChange.14h_RNA
# 1              2.4530557              5.9421698               7.308791
# 2 -> 3         0.1404357              0.7416598               1.635063
# 3 -> 2         0.6880281              2.4240733               4.066411
# 4             -0.1708422             -0.8515180              -1.514691
# 5             -0.5851530             -2.3090253              -3.493314

sig_any_cluster_id <- tbl_df(cbind(ensembl_gene_id = sig_any$ensembl_gene_id, cluster = sig_any_cluster_kmeans$kmeans$cluster)) 

# reassigning clusters such that 2 -> 3 and 3 -> 2 to make further plotting easier
sig_any_cluster_id <- sig_any_cluster_id %>% mutate(cluster =replace(cluster, cluster==2, 6)) %>% mutate(cluster =replace(cluster, cluster==3, 2)) %>% mutate(cluster =replace(cluster, cluster==6, 3))

# join the data_clusters with fold change data to enable plotting 
sig_any_cluster_id <- right_join(sig_any, sig_any_cluster_id)

#supp. table 2
write.csv(sig_any_cluster_id, file = "data/sig_any_cluster_id-5c-new.csv")

# line plots for clusters

sig_any_cluster_id_long <- sig_any_cluster_id %>% dplyr::select(external_gene_name, log2FoldChange.04h_RNA:log2FoldChange.14h_RNA, cluster) %>%pivot_longer(cols = log2FoldChange.04h_RNA:log2FoldChange.14h_RNA, names_to = "time", values_to = "foldchange")

sig_any_cluster_id_long_mean <- sig_any_cluster_id_long %>% 
        group_by(cluster, time) %>% 
        summarise(foldchange = mean(foldchange))

ggplot(data = sig_any_cluster_id_long,
mapping = aes(x = time, y = foldchange, group = external_gene_name, color = cluster)) +
geom_line(aes(group = external_gene_name), alpha = .1) +
geom_line(data = sig_any_cluster_id_long_mean, aes(group = 1), alpha = 1, color = "black", size = 1) +
facet_grid(cols = vars(cluster)) +
scale_color_viridis(discrete=TRUE, direction = 1) + geom_hline(yintercept=0, linetype="dashed", color = "grey60", size= 2) + mytheme + ylim(-5, 10)

# GO analysis
GO_analysis_cluster_final <- read_delim("data/GO-analysis-cluster-final.txt", "\t", escape_double = FALSE, trim_ws = TRUE)

names(GO_analysis_cluster_final) <- c("GOBP", "genes-in-group", "genes-in-data", "expected", "fold-enrichment", "p-val", "fdr", "neglog10FDR", "cluster")

GO_analysis_cluster_final$GOBP <- factor(GO_analysis_cluster_final$GOBP, levels = GO_analysis_cluster_final$GOBP)

GO_analysis_cluster_final$cluster <- as.character(GO_analysis_cluster_final$cluster)

ggplot(data = GO_analysis_cluster_final, aes(x = GOBP, y = neglog10FDR, fill = cluster)) + geom_bar(stat="identity", color = "black") + scale_fill_viridis(discrete=TRUE, direction = 1) + mytheme

# pdf(file = "02-data/plots/go-analysis.pdf",width = 9,height = 2)
# print(g3)
# dev.off()

```

##### Figure 3

```{r}
# input data
load("data/DE_RiboRNA_coding.RData")

dfa <- DE_RiboRNA_coding %>% select(gene_id:RiboRNA_padj, TPM_RNA:TPM_Ribo, condition, single_exon:gene_name_ok) %>% separate(gene_id, sep = "\\.", into = c("ensembl_gene_id", "extra")) %>%  mutate(target_status = ifelse(ensembl_gene_id %in% target_genes$Ensembl, "true", "false")) %>% mutate(significance = ifelse(Ribo_padj < 0.01, "true", "false"))  %>%
  mutate(up_down = ifelse(Ribo_log2FC > 0, "up", "down"))

colsa<-alpha(c("dark grey", "#042333"),c(.4,.7))
#names(colsa)<-c("target","non-target")

ggplot(dfa %>% arrange(target_status), 
            aes(
              x = log10(Ribo_baseMean), 
              y = Ribo_log2FC)) + 
  geom_point(aes(fill = target_status, color = target_status, pch = significance), size = 2)+ 
  facet_wrap(~as_factor(condition)) + 
  scale_fill_manual(values = colsa,"DUX4 target status") +
    scale_color_manual(values = colsa,"DUX4 target status") +
  scale_shape_manual(values = c(1, 21)) + 
  #theme(aspect.ratio = 1) +
  #ggtitle("DUX4") + 
  xlab(label = "log10 base expression") + 
  ylab(label = "log2 fold change DUX4/control")  + mytheme + geom_hline(yintercept=0, linetype="dashed", color = "black")

# table to use for n for various categories  
DE_table <- dfa %>% filter(significance == "true")
table(DE_table$up_down, DE_table$experiment, DE_table$target_status)

  #      time04h_vs_time00h time08h_vs_time00h time14h_vs_time00h
  # down                122               1273               2398
  # up                  194               1153               2288
  
table(DE_table$tx_type_RiboRNA, DE_table$experiment)

  #                 time04h_vs_time00h time08h_vs_time00h time14h_vs_time00h
  # TE_down                          1                 22                154
  # TE_up                            0                  5                 55
  # Concordant_down                113               1166               2122
  # Concordant_up                  177               1062               2077
  # mixed_NS                        25                171                278

# RNA vs ribo correlation  

dfa_te <- DE_RiboRNA_coding %>% select(gene_id:RiboRNA_padj, TPM_RNA:TPM_Ribo, condition, single_exon:gene_name_ok) %>% separate(gene_id, sep = "\\.", into = c("ensembl_gene_id", "extra")) %>%  mutate(target_status = ifelse(ensembl_gene_id %in% target_genes$Ensembl, "true", "false")) %>% mutate(significance = ifelse(RiboRNA_padj < 0.01, "true", "false")) 
colsi <- alpha(c("#440154","#3CBB75","gray43","gray10","dark gray"),c(.8,.8,.3,.3,.1))
names(colsi) <- c("TE_down","TE_up","Concordant_down","Concordant_up","mixed_NS")
axm <- max(abs(c(dfa$RNA_log2FC,dfa$Ribo_log2FC)),na.rm = T)

ggplot(dfa_te %>% arrange(desc(tx_type_RiboRNA)),
            aes(x = RNA_log2FC, y = Ribo_log2FC,
                color = tx_type_RiboRNA,
                fill = tx_type_RiboRNA,
                shape = significance)) + 
    geom_point(size = 2) + 
    mytheme +
    ylab(paste("Ribo-seq "," log2FC\n",sep = "")) +
    xlab(paste("RNA-seq log2FC\n",sep = "")) +
    scale_shape_manual(values = c(1, 21)) + 
    scale_color_manual(values = colsi,"Tx_class") +
    scale_fill_manual(values = colsi,"Tx_class") + 
  geom_hline(yintercept = 0,size=.5) + geom_vline(xintercept = 0,size=.5) + 
  coord_cartesian(xlim = c(-axm,axm),ylim=c(-axm,axm)) + 
  facet_wrap(.~experiment,nrow = 1) 

# table to use for n for various categories  
te_table <- dfa_te %>% filter(significance == "true")
table(te_table$tx_type_RiboRNA, te_table$experiment)

  #                 time04h_vs_time00h time08h_vs_time00h time14h_vs_time00h
  # TE_down                          1                 21                137
  # TE_up                            0                  2                 28
  # Concordant_down                  0                  0                  0
  # Concordant_up                    0                  0                  0
  # mixed_NS                         0                  0                  0

# exporting data for GO analysis
write.csv(te_table, file = "data/te_table.csv")

# step 1 - make the long table wide
DE_table_wide <- dfa[, c(22, 18, 9, 12, 14)]

cor(DE_table_wide %>% filter(condition == "time04h") %>% select(RNA_log2FC, Ribo_log2FC), use = "pairwise.complete.obs", method = "pearson")
# 0.7474313

cor(DE_table_wide %>% filter(condition == "time08h") %>% select(RNA_log2FC, Ribo_log2FC), use = "pairwise.complete.obs", method = "pearson")
# 0.8932658

cor(DE_table_wide %>% filter(condition == "time14h") %>% select(RNA_log2FC, Ribo_log2FC), use = "pairwise.complete.obs", method = "pearson")
# 0.9181838

# combining RNA/Ribo and SILAC data

DE_table_wide_14h <- DE_table_wide %>% filter(condition == "time14h")

load("data/data.rnaseq.silac.static_06072018.Rdata")
silac_data <- data.rnaseq.silac.static %>% select(geneid, Protein_log2FC = hl.ratio)

rna_ribo_protein_data <- left_join(DE_table_wide_14h, silac_data, by = c("gene_name_ok" = "geneid"))

cor(rna_ribo_protein_data %>% select(RNA_log2FC, Ribo_log2FC, Protein_log2FC), use = "pairwise.complete.obs", method = "pearson")

#                RNA_log2FC Ribo_log2FC Protein_log2FC
# RNA_log2FC      1.0000000   0.9186658      0.2987130
# Ribo_log2FC     0.9186658   1.0000000      0.3000128
# Protein_log2FC  0.2987130   0.3000128      1.0000000

# RNA/Ribo 0.919
# RNA/Protein 0.300
# Ribo/Protein 0.299

# using only genes with measurements in all three datasets
cor(rna_ribo_protein_data %>% select(RNA_log2FC, Ribo_log2FC, Protein_log2FC) %>% filter(complete.cases(.)), use = "pairwise.complete.obs", method = "pearson")

#NMD factor foldchange heatmap
nmd_genes <- tbl_df(c("UPF1", "UPF2", "UPF3B", "SMG1", "SMG5", "SMG6", "SMG7", "DCP1A", "DCP2", "XRN1", "RBM8A", "EIF4A3", "MAGOH", "CASC3", "GSPT1", "GSPT2", "ETF1")) 
names(nmd_genes) <- c("gene_name")

nmd_data <- left_join(nmd_genes, DE_table_wide_14h, by = c("gene_name" = "gene_name_ok"))
nmd_data <- left_join(nmd_data, silac_data, by = c("gene_name" = "geneid"))
write.csv(nmd_data, file = "data/nmd_factors_data.csv")
# replace SMG6 hl.ratio with NA
# read data in again
nmd_data <- read_csv("data/nmd_factors_data-mod.csv")

mat <- data.frame(nmd_data)
row.names(mat) <- nmd_data$gene_name
mat = mat[,c(4,5,7)]

mat_breaks = seq(-2, 2, by = 0.1)

pheatmap(t(mat),
color = cividis(length(mat_breaks) - 1),
breaks = mat_breaks, 
clustering_method = "ward.D2", #specify clustering method
clustering_distance_rows = "euclidean",
clustering_distance_columns = "euclidean",
border_color = "black",
na_col = "grey90", 
cluster_rows = F,
cluster_cols = F,
cellwidth = 12, cellheight = 12)
```

##### Figure S3

```{r}
load("data/QC_metacds.RData")

xinter<-c(1,20,70,170,216,236)
res_breaks <- xinter
res_labels <- c("TSS", "", "start\ncodon", "stop\ncodon", "", "TES")

ggplot(data=qc_metadf, 
             aes(x=position, y=mean,color=frame,fill=frame)) +
    geom_vline(xintercept=xinter, lty=3, color="black") +
    geom_bar(stat="identity") +
    geom_errorbar( aes(x=position, ymin=mean-sddevv, ymax=mean+sddevv), width=0.8, alpha=0.9) +
    scale_x_continuous(breaks=res_breaks, labels=res_labels) +
    labs(x="", y="P-sites coverage\n") +
    scale_color_manual(values = alpha(c("#440154","#FDE725","#29AF7F"),.8),"") +
    scale_fill_manual(values = alpha(c("#440154","#FDE725","#29AF7F"),.8),"") +
    mytheme + 
    facet_wrap(.~ condition,ncol = 2) + theme(legend.position = "none")

```

##### Figure 4
```{r}

load( "data/DE_ORFDEX.RData")

df<-ORFDEX

colsa<-alpha(c("#042333", "dark grey"),c(.7,.4))
names(colsa)<-c("NMD","canonical")

# M-A plot
ggplot(df %>% arrange(NMD_cand),
       aes(x=exonBaseMean+1, y=log2FC,
           color=NMD_cand,
           fill=NMD_cand)) + 
  geom_point(size = 2) +
  mytheme +
  scale_color_manual(values = colsa,"NMD status") +
  scale_fill_manual(values = colsa,"NMD status") +
  scale_shape_manual(values = c(1, 21)) + 
  facet_wrap(~condition,ncol = 1) + 
  scale_x_log10(breaks=c(1,11,101,1001,10001),labels=c(1,11,101,1001,10001)-1) +
  xlab(label = "log10 base expression") + 
  ylab(label = "log2 fold change DUX4/control")  + mytheme + geom_hline(yintercept=0, linetype="dashed", color = "black")


# boxplot
ggplot(df,aes(y=log2FC,x=NMD_cand,group=NMD_cand,fill=NMD_cand)) +
    geom_jitter(alpha=.06, pch = 20) +
    stat_summary(position=position_dodge(.6),size=.7, alpha = 1, pch= 21, color= "black") +
    scale_fill_manual(values = colsa,"ORF group") +
    theme_classic() +
    xlab("") +
    ylab("Ribo-seq log2FC") +
    theme(axis.title.x = element_text(size=22),axis.text.x  = element_text(angle=45, vjust=0.5, size=13)) +
    theme(axis.title.y = element_text(size=18),axis.text.y  = element_text(angle=0, vjust=0.5, size=15))  +
    theme(strip.text.x = element_text(size=10, face="bold"),strip.text.y = element_text(size=24),strip.background = element_rect(colour="black", fill="white")) + stat_compare_means(label="p.signif",method = "wilcox.test",hide.ns = F,comparisons = comps,label.y = 6) + facet_wrap(.~condition,ncol = 4) + coord_cartesian(ylim=c(-7,7)) +theme(axis.title.x=element_blank(),axis.text.x=element_blank(),
                                            axis.ticks.x=element_blank())

## GO analysis
nmd_exons_GO_analysis <- read_delim("data/nmd-exons-GO-analysis.txt",
"\t", escape_double = FALSE, trim_ws = TRUE)

names(nmd_exons_GO_analysis) <- c("GOBP", "genes-in-group", "genes-in-data", "expected", "fold-enrichment", "p-val", "fdr", "neglog10FDR")

nmd_exons_GO_analysis$GOBP <- factor(nmd_exons_GO_analysis$GOBP, levels = nmd_exons_GO_analysis$GOBP)

ggplot(data = nmd_exons_GO_analysis, aes(x = GOBP, y = neglog10FDR)) + geom_bar(stat="identity", fill = "#04233370") + mytheme

```

##### Figure 5

```{r}
#input data
polysome_traces <- read_excel("data/polysome-traces.xlsx")
names(polysome_traces) <- c("fraction", "0h", "4h", "8h", "14h")
polysome_traces_long <- polysome_traces %>% pivot_longer(-fraction, names_to = "time", values_to = "amount")

total_rna <- read_excel("data/total-rna.xlsx")
names(total_rna) <- c("fraction", "condition", "molecule", "amount")

colsa<-alpha(c("dark grey", "#042333"),c(1,1))
#names(colsa)<-c("target","non-target")

p1 <- ggplot(data = polysome_traces_long %>% filter(time %in% c("14h", "0h")), 
            aes(x = fraction, y = amount, color = time)) + 
  #geom_line(size = 1.5) + 
  #geom_point(size = 1) + 
geom_area(aes(fill = time), alpha = 0.3, position = "identity")+ 
  scale_fill_manual(values = colsa,"DUX4 target status") +
  scale_color_manual(values = colsa,"DUX4 target status") +
  theme_bw(base_size = 12) +
  #ggtitle("DUX4") + 
  ylab(label = "Absorbance 254 nm") + 
  xlab(label = "Fraction Number") +
  mytheme + theme(legend.position = "none")

p2 <- ggplot(data = (total_rna %>% filter(molecule == "RPL27")), 
            aes(x = fraction, y = amount, color = condition)) + 
  #geom_line(size = 1.5) + 
  #geom_point(size = 1) + 
geom_area(aes(fill = condition), alpha = 0.3, position = "identity")+ 
  scale_fill_manual(values = colsa,"DUX4 target status") +
  scale_color_manual(values = colsa,"DUX4 target status") +
  theme_bw(base_size = 12) +
  #ggtitle("DUX4") + 
  ylab(label = "% of RPL27 mRNA") + 
  xlab(label = "Fraction number")  +
  mytheme + theme(legend.position = "none")


p3 <- ggplot(data = (total_rna %>% filter(molecule == "DUX4")), 
            aes(x = fraction, y = amount, color = condition)) + 
geom_area(aes(fill = condition), alpha = 0.3, position = "identity")+ 
  scale_fill_manual(values = colsa,"DUX4 target status") +
  scale_color_manual(values = colsa,"DUX4 target status") +
  theme_bw(base_size = 12) +
  #ggtitle("DUX4") + 
  ylab(label = "% of DUX4 mRNA") + 
  xlab(label = "Fraction number")  +
  mytheme + theme(legend.position = "none")


p4 <- ggplot(data = (total_rna %>% filter(molecule == "ZSCAN4")), 
            aes(x = fraction, y = amount, color = condition)) + 
geom_area(aes(fill = condition), alpha = 0.3, position = "identity")+ 
  scale_fill_manual(values = colsa,"DUX4 target status") +
  scale_color_manual(values = colsa,"DUX4 target status") +
  theme_bw(base_size = 12) +
  #ggtitle("DUX4") + 
  ylab(label = "% of ZSCAN4 mRNA") + 
  xlab(label = "Fraction number")  +
  mytheme + theme(legend.position = "none")

p5 <- ggplot(data = (total_rna %>% filter(molecule == "SRSF3-Excl")), 
            aes(x = fraction, y = amount, color = condition)) + 
geom_area(aes(fill = condition), alpha = 0.3, position = "identity")+ 
  scale_fill_manual(values = colsa,"DUX4 target status") +
  scale_color_manual(values = colsa,"DUX4 target status") +
  theme_bw(base_size = 12) +
  #ggtitle("DUX4") + 
  ylab(label = "% of SRSF3-Excl mRNA") + 
  xlab(label = "Fraction number")  +
  mytheme + theme(legend.position = "none")

p6 <- ggplot(data = (total_rna %>% filter(molecule == "SRSF3-Incl")), 
            aes(x = fraction, y = amount, color = condition)) + 
geom_area(aes(fill = condition), alpha = 0.3, position = "identity")+ 
  scale_fill_manual(values = colsa,"DUX4 target status") +
  scale_color_manual(values = colsa,"DUX4 target status") +
  theme_bw(base_size = 12) +
  #ggtitle("DUX4") + 
  ylab(label = "% of SRSF3-Incl mRNA") + 
  xlab(label = "Fraction number")  +
  mytheme + theme(legend.position = "none")

legend_p7 <- ggplot(data = (total_rna %>% filter(molecule == "SRSF3-Incl")), 
            aes(x = fraction, y = amount, color = condition)) + 
geom_area(aes(fill = condition), alpha = 0.3, position = "identity")+ 
  scale_fill_manual(values = colsa,"DUX4 target status") +
  scale_color_manual(values = colsa,"DUX4 target status") +
  theme_bw(base_size = 12) +
  #ggtitle("DUX4") + 
  ylab(label = "% of SRSF3-Incl mRNA") + 
  xlab(label = "Fraction number")  +
  mytheme

legend_p7

fig1_nolegend <- plot_grid(p1, p2, p3, p4, p5, p6, 
                           #labels = c("B","C", "D", "E", "F", "G"), 
                           nrow = 3)

fig1_withlegend <- plot_grid(fig1_nolegend, legend, ncol = 1, rel_heights = c(1, .1))

fig1_nolegend
```
